backend/app/api/users/auth/services.py

```diff
@@ -0,0 +1,77 @@
+from jose import JWTError
+from sqlalchemy import or_
+from app.api.users.auth.utils import *
+
+
+async def user_registration_service(credentials, db):
+    input_sanitizer(credentials, db)
+    hashed_pass = hash_password(credentials.password)
+    new_user = User(
+        id=uuid4(),
+        email=credentials.email,
+        password=hashed_pass,
+        nickname=credentials.nickname,
+        score=0,
+        is_admin=False,
+        is_hidden=False,
+        is_verified=False,
+    )
+    db.add(new_user)
+    db.commit()
+    db.refresh(new_user)
+    return new_user
+
+
+def user_login_service(user_credentials, db):
+    user = db.query(User).filter(
+        or_(User.email == user_credentials.email_or_name, User.nickname == user_credentials.email_or_name)
+    ).first()
+    if not user:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail="User doesn't exist"
+        )
+    chk = verify_password(user_credentials.password, user.password)
```

Comment:
special case if user db password is null ?

Reply:
invalid si null non ?

Sinon on peut pas rentrer un password null 
`def password_validation(password: str):
    if len(password) < 6:
        return "Password length must be greater than 6 characters", False
    if not any(char.isdigit() for char in password):`

Reply:
si la personne a pas de password

Reply:
```py
    user = db.query(User).filter(
        or_(User.email == user_credentials.email_or_name, User.nickname == user_credentials.email_or_name),
        User.password is not None
    ).first()
```